tests/test_collectors.py

```diff
@@ -0,0 +1,268 @@
+) -> None:
+    """Integration test with a real orchestrator instance to ensure
+    output data matches expectations and proper db client API uage"""
+    entity = mock_entity(port=local_db.ports[0])
+
+    sink = mock_sink()
+    collector = DBMemoryCollector(entity, sink)
+
+    assert sink.num_saves == 0
+    await collector.prepare()
+    assert sink.num_saves == 1
+    await collector.collect()
+    assert sink.num_saves == 2
+
+    stats = sink.args
+    assert (
+        len(stats) == 4
+    )  # prove we filtered to expected data size, no timestamp in data.
+    assert stats["used_memory"] > 0  # prove used_memory was retrieved
+    assert stats["used_memory_peak"] > 0  # prove used_memory_peak was retrieved
+    assert stats["total_system_memory"] > 0  # prove total_system_memory was retrieved
+
+
+@pytest.mark.asyncio
+async def test_dbconncollector_collect(
+    mock_entity: MockCollectorEntityFunc,
+    mock_sink,
+    mock_redis,
+    mock_con,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Ensure that a valid response is returned as expected"""
+    entity = mock_entity()
+
+    sink = mock_sink()
+    collector = DbConnectionCollector(entity, sink)
+    with monkeypatch.context() as ctx:
+        ctx.setattr(redisa, "Redis", mock_redis(client_stats=mock_con(1, 2)))
+
+        assert sink.num_saves == 0
+        await collector.prepare()
+        assert sink.num_saves == 1
+        await collector.collect()
+        assert sink.num_saves == 3  # save twice w/two datapoints
+
+        stats = sink.args
+
+        idx = 1
+        id0, ip0 = f"ABC{idx}", f"127.0.0.{idx}:1234"
+        id1, ip1 = f"XYZ{idx}", f"127.0.0.{idx}:2345"
+        exp_clients = [{"id": id0, "addr": ip0}, {"id": id1, "addr": ip1}]
+
+        assert len(exp_clients) + 1 == len(stats)  # output includes timestamp
+        assert id0 in set(client["id"] for client in exp_clients)
+        assert id1 in set(client["id"] for client in exp_clients)
+        assert ip0 in set(client["addr"] for client in exp_clients)
+        assert ip1 in set(client["addr"] for client in exp_clients)
+
+
+@pytest.mark.asyncio
+async def test_dbconn_count_collector_collect(
+    mock_entity: MockCollectorEntityFunc,
+    mock_sink,
+    mock_redis,
+    mock_con,
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Ensure that a valid response is returned as expected"""
+    entity = mock_entity()
+
+    sink = mock_sink()
+    collector = DBConnectionCountCollector(entity, sink)
+    with monkeypatch.context() as ctx:
+        ctx.setattr(redisa, "Redis", mock_redis(client_stats=mock_con(1, 2)))
+
+        assert sink.num_saves == 0
+        await collector.prepare()
+        assert sink.num_saves == 1
+        await collector.collect()
+        assert sink.num_saves == 2
+
+        stats = sink.args
+        exp_counts = [{"num_clients": 1}]
+
+        assert len(exp_counts) + 1 == len(stats)  # output includes timestamp
```

Comment:
is this intended? From what I can tell `exp_counts` is of type `list` while `stats` is of type `dict`. Did we want to compare the length of these disparate types?

Reply:
I was thinking that the types don't really matter to the test - it only wants to verify that the things sent in were passed to the sink for persisting

Reply:
useless dict is useless. killing it off. i think i saw it as "illustrating what the result was" but it's just not useful.